expression/builtin_cast_vec.go

```diff
@@ -1272,11 +1272,47 @@ func (b *builtinCastStringAsRealSig) vecEvalReal(input *chunk.Chunk, result *chu
 }
 
 func (b *builtinCastStringAsDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastStringAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	if IsBinaryLiteral(b.args[0]) {
+		if err := b.args[0].VecEvalDecimal(b.ctx, input, result); err != nil {
+			return err
```

Comment:
Should we ``` return b.args[0].VecEvalDecimal(b.ctx, input, result)``` directly?  @qw4990  @wshwsh12 
Please add some test cases to cover such situations.

Reply:
Yes.

Reply:
> @ZhaoQian888 Friendly ping, please address comments.

I'm busy these days ,I will fix it when I have enough time